### PR TITLE
[ecma402] Proposes a Rust API for NumberFormat

### LINKED ISSUE #150: Implement ecma402 trait definition for `NumberFormat`

### PR DESCRIPTION
**Pull request by filmil**
`Intl.NumberFormat` is an ECMA402 API for formatting numbers, number with
units or with currencies based on international settings in CLDR.

Closes #150.

### CODE REVIEW

**Comment by filmil** [comment]
@zbraniecki FYI

**Comment by zbraniecki** [comment]
that looks good!

**Comment by filmil** [comment]
@kpozin PTAL?

File: ecma402_traits/src/numberformat.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            notation: options::Notation::Standard,
+            sign_display: options::SignDisplay::Auto,
+            style: options::Style::Decimal,
+            unit: None,
+            numbering_system: Default::default(),
+        }
+    }
+}
+
+use std::fmt;
```
Comment (kpozin) [inline ecma402_traits/src/numberformat.rs:181]: Nit: Consider moving this to the top of the scope.

**Pull request merged**